saltproc/app.py

```diff
@@ -53,52 +54,59 @@ def run():
         simulation.sim_depcode.write_runtime_input(msr,
                                                    step_idx,
                                                    simulation.restart_flag)
-        depcode.run_depletion_step(mpi_args, threads)
+
+        if rebuild_saltproc_results:
+            simulation.sim_depcode.rebuild_simulation_files(step_idx)
+        else:
+            depcode.run_depletion_step(mpi_args, threads)
         if step_idx == 0 and simulation.restart_flag is False:  # First step
             # Read general simulation data which never changes
-            simulation.store_run_init_info()
+            simulation.store_depcode_metadata()
             # Parse and store data for initial state (beginning of step_idx)
             mats = depcode.read_depleted_materials(False)
             simulation.store_mat_data(mats, step_idx - 1, False)
         # Finish of First step
         # Main sequence
         mats = depcode.read_depleted_materials(True)
         simulation.store_mat_data(mats, step_idx, False)
-        simulation.store_run_step_info()
+        simulation.store_step_neutronics_parameters()
+        simulation.store_step_metadata()
 
         # Reprocessing here
-        print("\nMass and volume of fuel before reproc: %f g, %f cm3" %
-              (mats['fuel'].mass,
-               mats['fuel'].vol))
-        # print("Mass and volume of ctrlPois before reproc %f g; %f cm3" %
-        #       (mats['ctrlPois'].mass,
-        #        mats['ctrlPois'].vol))
-        waste_streams, extracted_mass = reprocess_materials(mats,
-                                                            process_file,
-                                                            dot_file)
-        print("\nMass and volume of fuel after reproc: %f g, %f cm3" %
-              (mats['fuel'].mass,
-               mats['fuel'].vol))
-        # print("Mass and volume of ctrlPois after reproc %f g; %f cm3" %
-        #       (mats['ctrlPois'].mass,
-        #        mats['ctrlPois'].vol))
-        waste_and_feed_streams = refill_materials(mats,
-                                                  extracted_mass,
-                                                  waste_streams,
-                                                  process_file)
-        print("\nMass and volume of fuel after REFILL: %f g, %f cm3" %
-              (mats['fuel'].mass,
-               mats['fuel'].vol))
-        # print("Mass and volume of ctrlPois after REFILL %f g; %f cm3" %
-        #       (mats['ctrlPois'].mass,
-        #        mats['ctrlPois'].vol))
-        print("Removed mass [g]:", extracted_mass)
+        if not run_without_reprocessing:
```

Comment:
The double negative is confusing here. Usually, `if`s are clearer than `if not`s